### PR TITLE
Skip Early Depth Test When Unsupported in GLSL

### PR DESCRIPTION
**Pull request by zicklag**

### CODE REVIEW

File: src/back/glsl/features.rs [hunk 1]
```diff
@@ -206,7 +206,12 @@ impl<'a, W> Writer<'a, W> {
     /// [`Error::MissingFeatures`](super::Error::MissingFeatures) will be returned
     pub(super) fn collect_required_features(&mut self) -> BackendResult {
         if let Some(depth_test) = self.entry_point.early_depth_test {
-            self.features.request(Features::IMAGE_LOAD_STORE);
+            // If IMAGE_LOAD_STORE is supported for this version of GLSL
```
Comment (kvark) [inline src/back/glsl/features.rs:209]: let's just remove this check

File: src/back/glsl/mod.rs [hunk 2]
```diff
@@ -430,22 +430,27 @@ impl<'a, W: Write> Writer<'a, W> {
 
         // Enable early depth tests if needed
         if let Some(depth_test) = self.entry_point.early_depth_test {
-            writeln!(self.out, "layout(early_fragment_tests) in;")?;
+            // If early depth test is supported for this version of GLSL
+            if !(self.options.version < Version::Desktop(130)
```
Comment (kvark) [inline src/back/glsl/mod.rs:434]: let's do `log::warn!` in the `else` branch of this

File: src/back/glsl/features.rs (hunk 1 above)
Comment (zicklag) [reply src/back/glsl/features.rs:209]: On platforms where it is supported, we do check for the flag when writing out the extensions needed by the shader, does that mean we would still need the that?

File: src/back/glsl/mod.rs [hunk 3]
```diff
@@ -430,22 +430,27 @@ impl<'a, W: Write> Writer<'a, W> {
 
         // Enable early depth tests if needed
         if let Some(depth_test) = self.entry_point.early_depth_test {
-            writeln!(self.out, "layout(early_fragment_tests) in;")?;
+            // If early depth test is supported for this version of GLSL
```
[the remark this answers is not in the document]
Comment (zicklag) [reply src/back/glsl/mod.rs:433]: I like that!

**Review by Gordon-F — APPROVED** [review]

File: src/back/glsl/features.rs (hunk 1 above)
Comment (kvark) [reply src/back/glsl/features.rs:209]: sounds reasobable

**Review by kvark — APPROVED** [review]

**Pull request merged**